react/exercise/EX/Q3/src/view/containers/Main/Main.tsx

```diff
@@ -0,0 +1,204 @@
+        return {
+          ...state,
+          inputValues: newInputValue,
+        };
+      }
+      return state;
+    }
+    case ACTIONS.GUESS: {
+      if (inputValues.length === DIGITS) {
+        const lastResult = compare(inputValues, answer);
+        const newResults = [...results, lastResult];
+        const newButtonState = updateButtonState(lastResult, buttonState);
+        const isGameOver =
+          newResults.length === MAX_ATTEMPTS ||
+          isFullMatched(newResults[newResults.length - 1] || []);
+        const gameOverMessage = `正解は ${state.answer.join('')} でした`;
+        return {
+          ...state,
+          inputValues: [],
+          results: newResults,
+          buttonState: newButtonState,
+          isGameOver,
+          message: isGameOver ? gameOverMessage : message,
+        };
+      }
+      return state;
+    }
+
+    case ACTIONS.RESET: {
+      return {
+        ...initialGameState,
+        answer: generateAnswer(),
+      };
+    }
+  }
+};
+
+/* key 生成 */
+const generateKey = (col: number, row: number) => `c${col}r${row}`;
+
+const Main: FC = () => {
+  const [state, dispatch] = useReducer(reducer, initialGameState);
+  const { results, inputValues, buttonState, isGameOver, message } = state;
+
+  /* Methods */
+  //数字ボタンを押した時
+  const onPushNumber = useCallback((assignedKey: string) => {
+    dispatch({ type: ACTIONS.ADD_NUMBER, payload: assignedKey });
+  }, []);
+
+  // Guess ボタンを押した時
+  const onPushGuess = useCallback(() => {
+    dispatch({ type: ACTIONS.GUESS });
+  }, []);
+
+  // デリートボタンを押した時
+  const onPushDelete = useCallback(() => {
+    dispatch({ type: ACTIONS.DELETE_NUMBER });
+  }, []);
+
+  // リセットボタンを押した時
+  const onPushReset = useCallback(() => {
+    dispatch({ type: ACTIONS.RESET });
+  }, []);
+
+  // Tile コンポーネントに描画したい内容を(列/行)で取得するためのメソッド
+  const getTileInfo = (col: number, row: number): MatchResult => {
+    // results に存在していれば results から
+    if (results.length > row) {
+      return results[row][col];
+    }
+    // inputValues から取れそうであれば inputValues から
+    if (results.length === row) {
+      return {
+        value: inputValues[col] || undefined,
+      };
+    }
+    // まだ未操作ものであれば空を
+    return {};
+  };
+
+  // Tile コンポーネントをまとめて生成
+  const tiles = [...new Array(DIGITS * MAX_ATTEMPTS)].map((_, index) => {
+    const col = index % DIGITS;
+    const row = Math.floor(index / DIGITS);
+    const { value, type } = getTileInfo(col, row);
+    const selected = results.length === row && inputValues.length === col;
+    return (
+      <Tile
+        text={value}
+        type={type}
+        key={generateKey(col, row)}
+        selected={selected}
+      />
+    );
+  });
+
+  // Button コンポーネントをまとめて生成
+  const numButtons = useMemo(() => {
+    return [...new Array(RADIX)].map((_, index) => (
+      <NumberButton
+        assignedKey={`${index}`}
+        onPush={onPushNumber}
+        type={buttonState[index]}
+        disabled={isGameOver}
```

Comment:
@ka-clmx 
```suggestion
        disabled={isGameOver}
        key={index}
```
ここに key を入れないと console で警告が出ませんか？

Reply:
対処しました